pkg/controller/datavolume/snapshot-clone-controller.go

```diff
@@ -132,6 +132,10 @@ func (r *SnapshotCloneReconciler) addDataVolumeSnapshotCloneControllerWatches(mg
 		return err
 	}
 
+	if err := addDataSourceWatch(mgr, datavolumeController, "snapshotdatasource", dataVolumeSnapshotClone); err != nil {
```

Comment:
We should add some test coverage and ensure the detection pod works with the snapshot source. There might be something missing in this flow as the size-detection logic was implemented before this.

Reply:
Aw, cool. How come this worked when it was only implemented in the pvc source section?

Reply:
This watch is about data source changes, so not about size detection. Size detection for snap sources has been a thing for ages now